src/main/go/args/java_execution.go

```diff
@@ -6,14 +6,34 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"regexp"
 	"runtime"
+	"strconv"
 	"strings"
 )
 
 func getArchSpecificDirectory() string {
 	return fmt.Sprintf("%s-%s", runtime.GOOS, runtime.GOARCH)
 }
 
+func DetectJvmMajorVersion(javaBin string) (int, error) {
+	command := exec.Command(javaBin, "--version")
+	if err := command.Run(); err != nil {
+		return -1, err
```

Comment:
Wrap all errors in this function using `fmt.Errorf()`, in case the underlying ones are not informative enough. Lots of different things here can fail - executing java, matching regex, parsing values.